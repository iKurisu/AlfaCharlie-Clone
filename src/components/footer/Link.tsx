import React, { MouseEvent, RefObject } from "react";
import { Link, useHistory } from "react-router-dom";
import { connect } from "react-redux";
import useTransition from "hooks/useTransition";
import { easeInOut } from "utils/timings";
import { loaderActions } from "modules/loader";

interface MappedActions {
  toggleLoader: () => void;
}

interface OwnProps {
  to: string;
  footerRef: RefObject<HTMLDivElement>;
  backgroundRef: RefObject<HTMLDivElement>;
  fadeOutText: () => void;
}

type Props = MappedActions & OwnProps;

const FooterLink = ({
  to,
  footerRef,
  backgroundRef,
  fadeOutText,
  toggleLoader
}: Props): JSX.Element => {
  const history = useHistory();

  const bringToFront = useTransition(footerRef, {
    from: { zIndex: 200 },
    to: { zIndex: 200 },
    config: {
      duration: 0
    }
  });

  const expandBackground = useTransition(backgroundRef, {
    from: { height: "50%" },
    to: { height: "100%" },
    config: {
      duration: 1000,
      delay: 500,
      timing: easeInOut
    }
  });

  const click = (e: MouseEvent): void => {
    e.preventDefault();

    bringToFront();
    expandBackground();
    fadeOutText();
  };

  return <Link to={to} onClick={click} />;
};

const mapDispatch: MappedActions = {
  toggleLoader: loaderActions.toggleFooterLoader
};

export default connect(null, mapDispatch)(FooterLink);
