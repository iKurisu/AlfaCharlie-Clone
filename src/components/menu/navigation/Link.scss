.menu-nav-link {
  width: 100%;
  text-align: center;
  cursor: pointer;
  transition: opacity 0.2s;
}

.link-characters {
  font-family: 'benton-modern-display';
  font-size: 102px;
  letter-spacing: 6.3px;
  line-height: 1.1;
  color: #fff;
}

.menu-nav-links:hover .menu-nav-link:not(:hover) {
  opacity: 0.5;
}

@media (max-width: 480px) {
  .link-characters {
    font-size: 68px;
    letter-spacing: 4px;
  }
}

@media (max-width: 823px) and (orientation: landscape) {
  .link-characters {
    font-size: 49.5px;
    letter-spacing: 3px;
  }
}

@media (max-width: 1024px) {
  .link-characters {
    font-size: 102px;
    letter-spacing: 6.3px;
  }
}

@media (max-width: 1600px) {
  .link-characters {
    font-size: 95px;
    letter-spacing: 6px;
  }
}