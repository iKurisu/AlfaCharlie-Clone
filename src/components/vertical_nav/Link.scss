@import "sass/_variables.scss";

.vertical-nav-links li {
  display: inline-block;
  cursor: pointer;
}

.vertical-nav-links .link-wrapper {
  margin: 0 2vh;
  display: inline-block;
  position: relative;
  overflow: hidden;
}

.vertical-nav-links .link-name {
  display: block;
  margin: 0 1vh;
  color: $grey;
  text-transform: uppercase;
  font-size: 12px;
  letter-spacing: 7.05px;
  transition: color 0.3s;
  line-height: 10px;
}

.vertical-nav li:hover .link-name {
  color: $red;
}

.vertical-nav .active .link-name {
  color: $red;
}

.vertical-nav-links .link-mask {
  background: #3a3c49;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
}

@media (max-width: 1600px) {
  .vertical-nav-links .link-name {
    font-size: 10px;
    letter-spacing: 6px;
  }
}