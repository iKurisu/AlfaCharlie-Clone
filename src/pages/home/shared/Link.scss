.main-link {
  padding: 16px 0;
  display: inline-flex;
  align-items: center;
  position: relative;
  cursor: pointer;
}

.link-line {
  content: "";
  width: 23px;
  height: 1px;
  background: #d43246;
  display: inline-block;
  position: absolute;
  top: calc(50% - 1px);
  transform-origin: 0% 50%;
  transition: transform 0.35s cubic-bezier(0.645, 0.045, 0.355, 1) 0.33s;
}

.main-link svg {
  width: 36px;
  height: 11px;
  margin-top: -1px;
}

.main-link polyline,
.main-link line {
  fill: none;
  stroke: #d43246;
  stroke-width: 4;
}

.main-link polyline {
  stroke-dasharray: 60;
  stroke-dashoffset: 60;
  transition: stroke-dashoffset 0.3s;
}

.main-link line {
  stroke-dasharray: 140;
  stroke-dashoffset: 140;
  transition: stroke-dashoffset 0.4s;
}

.main-link .link-text-wrapper {
  display: inline-block;
  padding: 0 10px;
}

.main-link .link-text {
  color: #d43246;
  display: inline-block;
  font-size: 13px;
  letter-spacing: 0.16em;
  line-height: 1.1;
  text-transform: uppercase;
  transition: transform 1s cubic-bezier(0, 0, 0, 1) 0.15s;
}

.main-link:hover .link-line {
  transform: scaleX(0);
  transform-origin: 100% 50%;
  transition: transform 0.35s cubic-bezier(0.645, 0.045, 0.355, 1);
}

.main-link:hover polyline {
  stroke-dashoffset: 0;
  transition: stroke-dashoffset 0.3s 0.4s;
}

.main-link:hover line {
  stroke-dashoffset: 0;
  transition: stroke-dashoffset 0.3s 0.3s;
}

.main-link:hover .link-text {
  transform: translateX(5px);
  transition: transform 1s cubic-bezier(0, 0, 0, 1) 0.5s;
}

@media (max-width: 1600px) {
  .main-link .link-text {
    font-size: 12px;
  }
}
