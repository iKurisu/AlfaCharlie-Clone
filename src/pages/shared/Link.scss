.main-link {
  padding: 16px 0;
  display: inline-flex;
  align-items: center;
  position: relative;
  cursor: pointer;
}

.link-line {
  content: "";
  width: 23px;
  height: 1px;
  background: #d43246;
  display: inline-block;
  position: absolute;
  top: calc(50% - 1px);
  transform-origin: 0% 50%;
  transition: transform 0.35s cubic-bezier(0.645, 0.045, 0.355, 1) 0.33s;
}

.main-link svg {
  width: 36px;
  height: 11px;
  margin-top: -1px;
}

.main-link polyline,
.main-link line {
  fill: none;
  stroke: #d43246;
  stroke-width: 4;
}

.main-link polyline {
  stroke-dasharray: 60;
  stroke-dashoffset: 60;
  transition: stroke-dashoffset 0.3s;
}

.main-link line {
  stroke-dasharray: 140;
  stroke-dashoffset: 140;
  transition: stroke-dashoffset 0.4s;
}

.main-link .link-text-wrapper {
  display: inline-block;
  padding: 0 10px;
  white-space: nowrap;
}

.main-link .link-text,
.main-link .link-author,
.main-link .link-position {
  color: #d43246;
  font-size: 13px;
  letter-spacing: 0.16em;
  line-height: 1.1;
  text-transform: uppercase;
  transition: transform 1s cubic-bezier(0, 0, 0, 1) 0.15s;
}

.main-link .link-text {
  display: inline-block;
}

.main-link .link-author {
  padding-left: 4px;
}

.main-link .link-position {
  padding-left: 5px;
}

.main-link .link-position::before {
  content: "â€¢";
  margin: 0 6px;
}

.main-link:hover .link-line {
  transform: scaleX(0);
  transform-origin: 100% 50%;
  transition: transform 0.35s cubic-bezier(0.645, 0.045, 0.355, 1);
}

.main-link:hover polyline {
  stroke-dashoffset: 0;
  transition: stroke-dashoffset 0.3s 0.4s;
}

.main-link:hover line {
  stroke-dashoffset: 0;
  transition: stroke-dashoffset 0.3s 0.3s;
}

.main-link:hover .link-text {
  transform: translateX(5px);
  transition: transform 1s cubic-bezier(0, 0, 0, 1) 0.5s;
}

@media (max-width: 1600px) {
  .main-link .link-text,
  .main-link .link-author,
  .main-link .link-position {
    font-size: 12px;
  }
}

@media (max-width: 1024px) {
  .main-link .link-author {
    line-height: 1.2;
  }
}

@media (max-width: 768px), (max-width: 823px) and (orientation: landscape) {
  .main-link .link-author {
    padding: 0;
    line-height: 1;
  }

  .testimonials-content .link-text-wrapper {
    padding: 0 5px;
  }

  .testimonials-content .main-link svg {
    width: 30px;
  }
}

@media (max-width: 480px) {
  .main-link .link-author {
    padding: 0;
    line-height: 1.4;
  }

  .testimonials-content .link-text-wrapper {
    padding: 0 6.5px;
  }

  .testimonials-content .main-link svg {
    width: 36px;
  }
}
