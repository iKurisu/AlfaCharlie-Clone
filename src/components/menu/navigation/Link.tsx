import React from "react";
import "./Link.scss";

interface Props {
  link: string;
}

const Link = ({ link }: Props): JSX.Element => {
  return (
    <a className="menu-nav-link">
      <span className="link-characters">
        {link.split("").map(
          (character: string, id: number): JSX.Element => (
            <span className="link-character" key={id}>
              {character}
            </span>
          )
        )}
      </span>
    </a>
  );
};

export default Link;
