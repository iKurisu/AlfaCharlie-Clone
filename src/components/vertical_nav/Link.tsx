import React, { useRef, useEffect } from "react";
import useTransition from "hooks/useTransition";
import "./Link.scss";

interface Props {
  link: string;
  active: boolean;
  order: number;
  show: boolean;
  delay: number;
}

const Link = ({ link, active, order, show, delay }: Props): JSX.Element => {
  const mask = useRef(null);
  const name = useRef(null);

  const unmask = useTransition(mask, {
    from: { transform: "translateX(-101%)" },
    to: { transform: "translateX(101%)" },
    config: {
      delay: delay + order * 250,
      duration: 1400,
      timing: [0.85, 0, 0.15, 1]
    }
  });

  const fadeIn = useTransition(name, {
    from: { opacity: 0 },
    to: { opacity: 1 },
    config: {
      delay: delay + 700 + order * 200,
      duration: 700,
      timing: [0.55, 0, 0.45, 1]
    }
  });

  const fadeOut = useTransition(name, {
    from: { opacity: 1 },
    to: { opacity: 0 },
    config: {
      delay: order * 300,
      duration: 400,
      timing: [0.35, 0.6, 0.45, 1]
    }
  });

  useEffect((): void => {
    if (show) {
      unmask();
      fadeIn();
    } else {
      fadeOut();
    }
  }, [show]);

  return (
    <li>
      <span className={`link-wrapper${active ? " active" : ""}`}>
        <span className="link-name" ref={name} style={{ opacity: 0 }}>
          {link}
        </span>
        <span
          className="link-mask"
          style={{ transform: "translateX(-101%)" }}
          ref={mask}
        />
      </span>
    </li>
  );
};

export default Link;
