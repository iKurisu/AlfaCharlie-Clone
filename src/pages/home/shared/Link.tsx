import React from "react";
import "./Link.scss";

interface Props {
  text: string;
}

const Link = ({ text }: Props): JSX.Element => (
  <a className="main-link">
    <span className="link-line"></span>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
      version="1.1"
      x="0px"
      y="0px"
      viewBox="0 0 157.34 51.71"
      xmlSpace="preserve"
    >
      <polyline points="128.65,45.62 148.36,25.91 128.47,6.02 " />
      <line x1="8.46" y1="25.91" x2="147.73" y2="25.91" />
    </svg>
    <span className="link-text-wrapper">
      <span className="link-text">{text}</span>
    </span>
  </a>
);

export default Link;
