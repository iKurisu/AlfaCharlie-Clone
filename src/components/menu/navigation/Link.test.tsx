import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

it("renders correctly", (): void => {
  const component = renderer.create(<Link link="Agency" />);

  expect(component).toMatchSnapshot();
});
