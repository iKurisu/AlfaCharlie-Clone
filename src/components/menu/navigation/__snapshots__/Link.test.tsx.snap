// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="menu-nav-link"
>
  <span
    className="link-characters"
  >
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      A
    </span>
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      g
    </span>
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      e
    </span>
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      n
    </span>
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      c
    </span>
    <span
      className="link-character"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      y
    </span>
  </span>
</a>
`;
